sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +633,24 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): Option[VirtualJSFile] = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val formattedProps = javaOptions.collect {
+      case javaSysPropsPattern(propName, propValue) =>
+        "\"" + escapeJS(propName) + "\": \"" + escapeJS(propValue) + "\""
+    }
```

Comment:
My point is: *where* should it fail? And how would that place make sure it understands all "handlers"?